### PR TITLE
Tabular: Added 'original' feature importance

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

Changes:
- Added 'original' feature importance
     - This is the original features only, so automatically generated ngrams are not included, and it will properly give importance calculations to text features.
- Updated raw arg to feature_stage for feature importance
- Lowered default subsample_size to 1,000 from 10,000
- Fixed crash in feature importance when sent a dataset with nan class labels after label transformation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-479-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-479/1/index.html

File: autogluon/task/tabular_prediction/predictor.py [hunk 1]
```diff
@@ -501,23 +501,32 @@ def feature_importance(self, model=None, dataset=None, features=None, raw=True,
...
         features : list, default = None
             List of str feature names that feature importances are calculated for and returned, specify None to get all feature importances.
             If you only want to compute feature importances for some of the features, you can pass their names in as a list of str.
-        raw : bool, default = True
-            Whether to compute feature importance on raw features in the original data (after automated feature engineering) or on the features used by the particular model.
-            For example, a stacker model uses both the original features and the predictions of the lower-level models.
-            Note that for bagged models, feature importance calculation is not yet supported when both `raw=True` and `dataset=None`. Doing so will raise an exception.
-        subsample_size : int, default = 10000
+        feature_stage : str, default = 'original'
+            The stage of features to compute importances on.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:517]: Can we clarify: "The stage of features to compute importances on"
 -> 
"What stage of feature-processing should importances be computed for."

File: autogluon/task/tabular_prediction/predictor.py [hunk 2]
```diff
@@ -501,23 +501,32 @@ def feature_importance(self, model=None, dataset=None, features=None, raw=True,
...
+            Options:
+                'original':
+                    Compute importances of the original features.
+                    Warning: `dataset` must be specified with this option, otherwise an exception will be raised.
+                'transformed':
+                    Compute importances of the post-internal-transformation features (after automated feature engineering). These features may be missing some original features, or add new features entirely.
+                    An example of new features would be ngram features generated from a text feature.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:524]: "text feature" -> "text column"

File: autogluon/task/tabular_prediction/predictor.py [hunk 3]
```diff
@@ -501,23 +501,32 @@ def feature_importance(self, model=None, dataset=None, features=None, raw=True,
...
             This dataset must also contain the label-column with the same column-name as specified during fit().
             If specified, then the dataset is used to calculate the feature importance scores.
             If str is passed, `dataset` will be loaded using the str value as the file path.
             If not specified, the original dataset used during fit() will be used if `cache_data=True`. Otherwise, an exception will be raised.
             Do not pass the training data through this argument, as the feature importance scores calculated will be inaccurate.
+        model : str, default = None
+            Model to get feature importances for, if None the best model is chosen.
+            Valid models are listed in this `predictor` by calling `predictor.model_names`
         features : list, default = None
             List of str feature names that feature importances are calculated for and returned, specify None to get all feature importances.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:514]: Docs should clarify how the options of possible feature names change depending on what user specified for `feature_stage`

File: autogluon/task/tabular_prediction/predictor.py [hunk 4]
```diff
@@ -501,23 +501,32 @@ def feature_importance(self, model=None, dataset=None, features=None, raw=True,
...
         ----------
-        model : str, default = None
-            Model to get feature importances for, if None the best model is chosen.
-            Valid models are listed in this `predictor` by calling `predictor.model_names`
         dataset : str or :class:`TabularDataset` or `pandas.DataFrame` (optional)
             This dataset must also contain the label-column with the same column-name as specified during fit().
             If specified, then the dataset is used to calculate the feature importance scores.
             If str is passed, `dataset` will be loaded using the str value as the file path.
             If not specified, the original dataset used during fit() will be used if `cache_data=True`. Otherwise, an exception will be raised.
             Do not pass the training data through this argument, as the feature importance scores calculated will be inaccurate.
```
Comment (jwmueller) [inline autogluon/task/tabular_prediction/predictor.py:509]: As inaccurate might sound vague, may be good to clarify: 
"inaccurate." 
-> 
"biased due to overfitting.  More accurate feature-importances will be obtained from new data that was held-out during `fit()`."

File: autogluon/task/tabular_prediction/predictor.py (hunk 1 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py]: Added

File: autogluon/task/tabular_prediction/predictor.py (hunk 2 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py]: Added

File: autogluon/task/tabular_prediction/predictor.py (hunk 3 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py:515]: Added

File: autogluon/task/tabular_prediction/predictor.py (hunk 4 above)
Comment (Innixma) [reply autogluon/task/tabular_prediction/predictor.py]: Added

**Comment by Innixma** [comment]
> Looks good, just have minor docs comments. Should we add small unit test for feature_importance as well?

Added advanced functionality unit test

**Comment by szha** [comment]
Job PR-479-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-479/2/index.html

**Pull request merged**